File: 11_JavaScript/1_Basico/4_poo/1_bassico/CursoPOOUber/JS/Car.js
class Car {

    constructor(license,driver) {
        this.id;
        this.license = license;
        this.driver = driver;
        this.passegenger;
    }

    printDataCar() {
        console.log(`License: ${this.license}`);
        console.log(`Name: ${this.driver.name}`);
        console.log(`Document: ${this.driver.document}`);
    }
}

File: 11_JavaScript/1_Basico/4_poo/1_bassico/CursoPOOUber/Java/Car.java
class Car {
    Integer id;
    String license;
    Account driver;
    private Integer passegenger;

    public Car(String license, Account driver){
        this.license = license;
        this.driver = driver;
        // passegenger = 3;
        // System.out.println("Passenger: " + passegenger);
    }

    void printDataCar() {
        if(passegenger != null) {
            System.out.println("License: " + license + " / Driver: " + driver.name + " / Passenger: " + passegenger);
        }
    }

    public Integer getPassenger() {
        return passegenger;
    }

    public void setPassenger(Integer passenger) {
        if(passenger == 4) {
            this.passegenger = passenger;
        } else {
            System.out.println("Necesitas Asignar 4 pasajeros");
        }
    }

}
